os/fs/smartfs/smartfs_smart.c

```diff
@@ -1805,19 +1805,19 @@ static int smartfs_mkdir(struct inode *mountpt, const char *relpath, mode_t mode
 		/* Allocate new data sector for entry */
 		ret = smartfs_alloc_firstsector(fs, &entry.firstsector, SMARTFS_DIRENT_TYPE_DIR, NULL);
 		if (ret != OK) {
-			fdbg("Failed to allocate data sector for entry\n");
+			fdbg("Failed to allocate data sector for new entry, error = %d\n", ret);
 			goto errout_with_semaphore;
 		}
 		/* Try to find empty/invalid entry available in one of he chained parent sectors */
 		ret = smartfs_find_availableentry(fs, &entry);
 		if (ret != OK) {
-			fdbg("find_availableentry failed, cannot find entry for writing\n");
+			fdbg("find_availableentry failed, cannot find entry for writing, error = %d\n", ret);
 			goto errout_with_semaphore;
 		}
 		/* Now we have an entry allocated for writing, write new entry to sector */
 		ret = smartfs_writeentry(fs, entry, SMARTFS_DIRENT_TYPE_DIR, mode);
 		if (ret != OK) {
-			fdbg("Failed to write new entry to sector\n");
+			fdbg("Failed to write new entry to MTD, error = %d\n", ret);
```

Comment:
you don't have to mention MTD here, this can be attached to block driver if smart.c expanded. MTD is managed by smart.c

Reply:
Done